docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+Parts
+=====
+
+In the Craft Parts framework, parts are descriptions of the components to be
+build for deployment, either individually or as part of a larger build
+containing many components.
+
+Tools like `Snapcraft`_ and `Juju`_ that use the concepts of parts to
+describe a build process typically accept specifications of parts in YAML format. This allows each part to be described in a convenient,
+mostly-declarative format.
+
+When the Craft Parts framework is used to process a part on behalf of a tool
+or library, it performs the steps described in the
+:ref:`parts lifecycle <lifecycle>` to *pull* the source code and dependencies, *overlay* them onto a base file system, *build* the part, then *organize*,
+*stage* and *prime* the build products for further processing.
+
+Describing a part
+-----------------
+
+Each part contains all the required information about a specific component,
+defining pieces of information using keys and values, as in a dictionary.
+It can also be thought of as an object with named properties.
+
+Generally, each part includes information about the following:
+ * Its `source <Source_>`_
+ * Its `dependencies <Dependencies_>`_
+ * The `build process <Build process_>`_
+ * How `build products <Build products_>`_ are exported/installed
+
+Each of these are described in the following sections.
+
+In addition, if more than one part is specified, the order in which the parts
+are built is also partially defined by its description. See `Defining the build order`_ for more information.
+
+.. ### Link to a schema or complete overview in the reference section.
+
+Source
+~~~~~~
+
+The source for a part is described using the ``source`` key. This
+specifies a location where the source code or other information is to be
+*pulled* from, and may be a repository on a remote server, a directory on
+the build host, or some other location.
+
+Additional keys are used to fine-tune the specification so that a precise
+description of the source location can be given, and also to specify the type
+of source to be processed.
+
+Where the type of the source information cannot be automatically determined,
+the ``source-type`` key is used to explicitly specify the source format.
+This influences the way in which the source code or data is processed.
+A list of supported formats can be found in the `craft_parts/sources/sources.py` file. Examples include "bzr", "git", "hg", "mercurial", "subversion", "svn", "tar", "zip", "deb", "rpm", "7z", "local".
```

Comment:
opinion: This list doesn't read well (lots of quotes and commas) and is not alphabetical.   I would either put these as a bullet points, remove the quotes, or just choose 3 or 4 popular formats to mention here.

Reply:
Thanks. This was from a dump of the old Snapcraft schema. I'll choose a few and link out to a more comprehensive list.